test/unit/input_template_renderer_test.rb

```diff
@@ -95,6 +95,147 @@
+          template.template_inputs << FactoryGirl.build(:template_input, :name => 'debug', :input_type => 'user')
+        end
+      end
+
+      let(:package_template) do
+        FactoryGirl.build(:job_template, :name => 'package action', :template => <<TEMPLATE)  do |template|
+<%= render_template("command action", "command" => "yum -y \#{ input("action") } \#{ input('package') }") -%>
+TEMPLATE
+          template.template_inputs << FactoryGirl.build(:template_input, :name => 'package', :input_type => 'user')
+          template.template_inputs << FactoryGirl.build(:template_input, :name => 'action', :input_type => 'user')
+          template.foreign_input_sets << FactoryGirl.build(:foreign_input_set, :target_template => command_template, :include_all => true, :exclude => 'command')
+        end
+      end
+
+      let(:template) do
+        FactoryGirl.create(:job_template,
+                           :template => '<%= render_template("package action", "action" => "install", :with_foreign_input_set => true) %>').tap do |template|
+          template.foreign_input_sets << FactoryGirl.build(:foreign_input_set, :target_template => package_template, :include_all => true, :exclude => 'action')
+        end
+      end
+
+      let(:job_invocation) { FactoryGirl.create(:job_invocation) }
+      let(:template_invocation) { FactoryGirl.build(:template_invocation, :template => template) }
+      let(:renderer) { InputTemplateRenderer.new(template) }
+      let(:result) { renderer.render }
+
+      before do
+        User.current = users :admin
+        command_template.save!
+        package_template.save!
+        job_invocation.template_invocations << template_invocation
+      end
+
+      describe 'foreign input set' do
+        describe 'with include_all' do
+          let(:template) do
+            FactoryGirl.create(:job_template, :template => '<%= render_template("package action", "action" => "install) %>').tap do |template|
+              template.foreign_input_sets << FactoryGirl.build(:foreign_input_set, :target_template => package_template, :include_all => true)
+            end
+          end
+
+          it 'includes all inputs from the imported template' do
+            template.template_inputs_with_foreign.map(&:name).sort.must_equal ['action', 'debug', 'package']
+          end
+        end
+
+        describe 'with include_all and some excludes' do
+          let(:template) do
+            FactoryGirl.create(:job_template, :template => '<%= render_template("package action", "action" => "install") %>').tap do |template|
+              template.foreign_input_sets << FactoryGirl.build(:foreign_input_set, :target_template => package_template, :include_all => true, :exclude => 'action,debug')
+            end
+          end
+
+          it 'includes all inputs from the imported template except the listed once' do
+            template.template_inputs_with_foreign.map(&:name).sort.must_equal ['package']
```

Comment:
could you please add the same test but with both include and exclude set to empty string (UI vs API)